theano/gof/cc.py

```diff
@@ -287,21 +287,23 @@ def struct_gen(args, struct_builders, blocks, sub):
             %(struct_init_head)s
             this->__ERROR = __ERROR;
             return 0;
-        }
+        }// end init()
         void cleanup(void) {
             %(struct_cleanup)s
             %(storage_decref)s
-        }
-        int run(void) {
+        } // end cleanup()
+        int run(void);
+    };
+    """ % sub
+    run_code = """
+        DllExport int %(name)s::run(void) {
```

Comment:
I made a commit to remove the  DllExport for non c_callable function. But now I think that  is no enough . Probably we should make refactor the code so that for non c_callable function we can use a anonymous namespace.